### PR TITLE
FISH-6195 Add extra logging and remove redundant logger

### PR DESCRIPTION
**Pull request by Pandrex247**
The redundant logger, `Logger LOGGER = Logger.getLogger(CLICommand.class.getPackage().getName())`, is the same as the one inherited from the `CLICommand` class itself, `logger`.

To log messages at level `FINER`, set the environment variable `AS_DEBUG` to _true_, to log messages at level `FINEST`, set the environment variable `AS_TRACE` to _true_.

Signed-off-by: Andrew Pielage <pandrex247@hotmail.com>

### CODE REVIEW

File: src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java [hunk 1]
```diff
@@ -125,9 +121,11 @@ protected void validate() throws CommandException {
 
         // Set up the install dir variable
         glassfishDir = getInstallRootPath();
+        logger.log(Level.FINEST, "glassfishDir resolved as {0}", glassfishDir);
```
Comment (aubi) [inline src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java:124]: Using inherited logger will report wrong file in the log.
I see the usage even in LocalDomainCommand, it is wrong as well.

File: src/main/java/fish/payara/extras/upgrade/UpgradeServerCommand.java [hunk 2]
```diff
@@ -329,7 +331,7 @@ private void validateDistribution() throws CommandValidationException {
         // Check if distribution is defined.
         // Continue with upgrade if no defined distribution, user should be able rollback if needed.
         if (versionDistribution.isEmpty()) {
-            System.out.println("The distribution cannot be validated.");
+            logger.log(Level.INFO, "The distribution cannot be validated.");
```
Comment (aubi) [inline src/main/java/fish/payara/extras/upgrade/UpgradeServerCommand.java:334]: Is the INFO level appropriate? I understand the message as more important, System.out was always printed, so I would expect SEVERE...

File: src/main/java/fish/payara/extras/upgrade/UpgradeServerCommand.java [hunk 3]
```diff
@@ -738,7 +796,13 @@ private void fixNadminPermissions() throws IOException {
             }
 
             if (nadminPath.toFile().exists()) {
+                logger.log(Level.FINER, "Fixing file permissions for {0} to {1}",
+                        new Object[]{nadminPath.toString(), "rwxr-xr-x"});
```
Comment (aubi) [inline src/main/java/fish/payara/extras/upgrade/UpgradeServerCommand.java:800]: I would suggest to move "rwxr-xr-x" to either variable or constant.

**Review by aubi — CHANGES_REQUESTED** [review]
I like the amount of logs.
I like moving from LOGGER to logger (as it is not a constant), but it must be a private field. The inherited logger will report file CLICommand.java.

**Review by fturizo — APPROVED** [review]
The full PR and logging changes look good, @Pandrex247. However, from a user experience perspective, asking users to modify the `AS_DEBUG` and `AS_TRACE` environment variables prior to a server upgrade might be a bit cumbersome.

Would it be possible to implement an argument like `--verbose` or `--debug`/`--trace`that can automatically set the logger levels at the very start of the command? This way users don't have to worry about setting up the variables and the command can be run on a seamless manner.

File: src/main/java/fish/payara/extras/upgrade/UpgradeServerCommand.java (hunk 2 above)
Comment (Pandrex247) [reply src/main/java/fish/payara/extras/upgrade/UpgradeServerCommand.java:334]: I'll change it to WARNING - SEVERE to me implies something has gone critically wrong, this message is just a "I don't know what you are so I'm going in blind" message

**Comment by Pandrex247** [comment]
> The full PR and logging changes look good, @Pandrex247. However, from a user experience perspective, asking users to modify the `AS_DEBUG` and `AS_TRACE` environment variables prior to a server upgrade might be a bit cumbersome.
> 
> Would it be possible to implement an argument like `--verbose` or `--debug`/`--trace`that can automatically set the logger levels at the very start of the command? This way users don't have to worry about setting up the variables and the command can be run on a seamless manner.

Having a `--verbose` or `--debug` option change the logger level would make this the only command where doing so does this. In terms of Payara those terms essentially mean something else. 
It's poorly documented but environment or system property is how debug logging is enabled for CLI Commands and I don't think we should change that behaviour on a command-by-command basis.

File: src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java (hunk 1 above)
Comment (aubi) [reply src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java:124]: The log output is correct, it just doesn't follow logger setup for this class or package, it follows rules for com.sun.enterprise.admin.servermgmt.cli.
But if it is done this way for all commands, let's discuss it elsewhere.

**Review by aubi — APPROVED** [review]
LGTM

**Pull request merged**